gw_eccentricity/eccDefinition.py

```diff
@@ -251,16 +296,25 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
         self.t_for_omega22_average = None
         self.omega22_average = None
 
-        if "hlm_zeroecc" in self.dataDict:
-            self.compute_res_amp_and_omega22()
+        # compute residual data
+        if any(["hlm_zeroecc" in self.dataDict,
+                "amplm_zeroecc" in self.dataDict and "omegalm_zeroecc" in self.dataDict,
+                "amplm_zeroecc" in self.dataDict and "phaselm_zeroecc" in self.dataDict]):
+            self.compute_res_amp22_and_res_omega22()
```

Comment:
Why does this need a separate function?

Reply:
What do you mean? computing the residual amp22 and omega22 is lengthy, we did not want to do all those staff in init itself and decided to do it in a separate function and call it in init.